test/realm_test.dart

```diff
@@ -1353,20 +1353,43 @@ Future<void> main([List<String>? args]) async {
     expect(transferredBytes, greaterThan(-1));
   });
 
-  baasTest('Realm.open (flexibleSync) - download a populated realm', (appConfiguration) async {
+  baasTest('Realm.open async (flexibleSync) with initial subscriptions - download a populated realm', (appConfiguration) async {
     final app = App(appConfiguration);
     final queryDifferentiator = generateRandomString(10);
-    const itemCount = 200;
-    final config = await _addDataToAtlas(app, queryDifferentiator: queryDifferentiator, itemCount: itemCount);
+    final itemCount = 200;
+    await _addDataToAtlas(app, queryDifferentiator: queryDifferentiator, itemCount: itemCount);
+    final user = await app.logIn(Credentials.anonymous(reuseCredentials: false));
+    final config = Configuration.flexibleSync(
+      user,
+      [Product.schema],
+      initialSubscriptionsConfiguration: InitialSubscriptionsConfiguration((realm) {
+        realm.subscriptions.update((mutableSubscriptions) {
+          mutableSubscriptions.add(realm.all<Product>());
+        });
+      }),
+    );
+
     var syncedRealm = await getRealmAsync(config);
     expect(syncedRealm.isClosed, false);
     final data = syncedRealm.query<Product>(r'stringQueryField BEGINSWITH $0', [queryDifferentiator]);
     expect(data.length, itemCount);
   });
 
-  baasTest('Realm.open (flexibleSync) - listen for download progress of a populated realm', (appConfiguration) async {
+  baasTest('Realm.open async (flexibleSync) with initial subscriptions - listen for download progress', (appConfiguration) async {
     final app = App(appConfiguration);
-    final config = await _addDataToAtlas(app);
+    final queryDifferentiator = generateRandomString(10);
+    final itemCount = 100;
+    await _addDataToAtlas(app, queryDifferentiator: queryDifferentiator, itemCount: itemCount);
```

Comment:
Can we add this data while we are creating the apps? Seems we need this data on many tests now. And each test will be adding this data over and over again

Reply:
Adding data takes time and will slow down the tests execution. Also the data that I add have dfferent Product name for each test.

Reply:
I agree - we should not have test data added as part of setting up the apps because tests should not depend on each other - imagine having one test that deletes items from the test data, then another test that requires there are 100 elements. Then the second test will pass or fail depending on the order in which tests are run.